File: src/main/java/com/minhyun/quydu_be/entity/Review.java
package com.minhyun.quydu_be.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

@Entity
@Table(name = "reviews")
public class Review extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "order_id", nullable = false)
    private Order order;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "listing_id", nullable = false)
    private Listing listing;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "seller_id", nullable = false)
    private User seller;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "buyer_id", nullable = false)
    private User buyer;

    @Column(nullable = false)
    private Integer rating;

    @Column(length = 2000)
    private String comment;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private ReviewStatus status = ReviewStatus.PENDING;

    public Order getOrder() { return order; }
    public void setOrder(Order order) { this.order = order; }
    public Listing getListing() { return listing; }
    public void setListing(Listing listing) { this.listing = listing; }
    public User getSeller() { return seller; }
    public void setSeller(User seller) { this.seller = seller; }
    public User getBuyer() { return buyer; }
    public void setBuyer(User buyer) { this.buyer = buyer; }
    public Integer getRating() { return rating; }
    public void setRating(Integer rating) { this.rating = rating; }
    public String getComment() { return comment; }
    public void setComment(String comment) { this.comment = comment; }
    public ReviewStatus getStatus() { return status; }
    public void setStatus(ReviewStatus status) { this.status = status; }
}

File: src/types/review.ts
import type { OrderStatus } from "./order";

export type ReviewStatus = "PENDING" | "APPROVED" | "EDITED" | "HIDDEN";

export type Review = {
  id: string;
  orderId: string;
  listingId: string;
  sellerId: string;
  buyerId: string;
  rating: number; // 1–5
  comment?: string;
  status: ReviewStatus;
  createdAt?: string; // ISO
  updatedAt?: string; // ISO
};

export type CreateReviewRequest = {
  orderId: string;
  listingId: string;
  sellerId: string;
  rating: number;
  comment?: string;
};

export type AdminUpdateReviewRequest = {
  rating?: number;
  comment?: string;
  status?: ReviewStatus;
};

